### PR TITLE
Update for mono repo

### PR DESCRIPTION
**Pull request by slesaad**
# Description
Changes to work with `veda-deploy` + others discovered along the way

# Changes
- Added composite deploy action that can be used by `veda-deploy` as well as `cicd` in this repo
- Made a lot of env vars optional (assume roles, vector info, etc)
- Update env var names to work together with other repo for mono repo deployment - https://github.com/NASA-IMPACT/veda-data-airflow/issues/112
- Bring in the fix for https://github.com/US-GHG-Center/ghgc-data-airflow/issues/20, from ghgc project https://github.com/US-GHG-Center/ghgc-data-airflow/pull/19
- Added support for permissions boundary for MCP
- Added secret access to workflows api lambda

# Prerequisites
https://github.com/NASA-IMPACT/veda-auth/pull/27

# TODO
- [x] Permission to workflows lambda to read cognito secret

### CODE REVIEW

**Comment by smohiudd** [comment]
@slesaad why is there such a big diff on this PR? It looks like the `update-for-mono-repo` was far behind `dev`

File: workflows_api/runtime/src/config.py
Comment (slesaad) [inline workflows_api/runtime/src/config.py:16]: don't need client_id/client_secret since we're differing the token generation to `veda-backend`

`userpool_id` and `jwks_url` can be derived from `workflows_client_secret_id`

File: workflows_api/runtime/src/auth.py
Comment (slesaad) [inline workflows_api/runtime/src/auth.py:121]: unused

File: workflows_api/runtime/src/auth.py
Comment (slesaad) [inline workflows_api/runtime/src/auth.py:28]: instead of asking for `jwks_url` env var, derive from secrets

File: infrastructure/variables.tf
Comment (slesaad) [inline infrastructure/variables.tf:77]: append `veda_` to work in sync with other repos through `veda-deploy`

File: infrastructure/main.tf
Comment (slesaad) [inline infrastructure/main.tf:51]: these lines were added to allow deploying without `vector_vpc`being required

File: infrastructure/main.tf
Comment (slesaad) [inline infrastructure/main.tf:6]: making optional

File: infrastructure/custom_policies/main.tf
Comment (slesaad) [inline infrastructure/custom_policies/main.tf:85]:
```suggestion
      "*"
```

File: infrastructure/custom_policies/main.tf
Comment (slesaad) [inline infrastructure/custom_policies/main.tf:71]: allow assuming roles if arns provided, else deny

File: docker_tasks/cogify_transfer/handler.py
Comment (slesaad) [inline docker_tasks/cogify_transfer/handler.py:59]: making optional

File: docker_tasks/build_stac/utils/stac.py
Comment (slesaad) [inline docker_tasks/build_stac/utils/stac.py:64]: should probably not do this 🤔

File: docker_tasks/build_stac/utils/stac.py
Comment (slesaad) [inline docker_tasks/build_stac/utils/stac.py:24]: making optional

**Comment by slesaad** [comment]
> @slesaad why is there such a big diff on this PR? It looks like the `update-for-mono-repo` was far ahead `dev`
> 
> What were the `veda-deploy` related changes?

a lot of things were discovered when deploying to mcp and in another stack, like a lot of things that needed to be optional were not, among other things. i've added comments to most of the changes. looks like there's one change that sneaked in that we don't want that i'll update

File: .github/workflows/cicd.yml
Comment (anayeaye) [inline .github/workflows/cicd.yml:70]: Do we lose our ability to deploy to UAH here where we do not have a deployment role? Or have we never had github push actions for cycling changes to the dev and staging MWAAs in UAH?

Comment (slesaad) [reply .github/workflows/cicd.yml:70]: yep, the current workflow https://github.com/NASA-IMPACT/veda-data-airflow/blob/dev/.github/workflows/deploy.yml already uses it

File: workflows_api/runtime/src/auth.py
Comment (smohiudd) [reply workflows_api/runtime/src/auth.py:28]: auth secrets doesn't return `SecretString` anymore but is a dict. Will need:
```
aws_region = response['aws_region']
userpool_id = response['userpool_id']

return f"https://cognito-idp.{aws_region}.amazonaws.com/{userpool_id}/.well-known/jwks.json"
```

File: infrastructure/terraform.tfvars.tmpl
Comment (slesaad) [inline infrastructure/terraform.tfvars.tmpl:8]:
```suggestion
cognito_app_secret="${VEDA_PROGRAMMATIC_CLIENT_SECRET_ID}"
```

Comment (slesaad) [reply infrastructure/terraform.tfvars.tmpl:8]: i agree that `cognito_app_secret` is confusing, but that's just internally used. The env var name is gonna be `VEDA_PROGRAMMATIC_CLIENT_SECRET_ID` - coming from `veda-auth` via `veda-deploy`.

Comment (slesaad) [reply infrastructure/terraform.tfvars.tmpl:8]: > [!CAUTION]
Note to update the repository secrets before merging to dev.

**Comment by anayeaye** [comment]
I have added to the dev environment secrets used in the github **dev** workflow
`VEDA_WORKFLOWS_CLIENT_SECRET_ID` contains the id of the aws secrets manager containing the workflows client config
`VEDA_PROGRAMMATIC_CLIENT_SECRET_ID` contains the same for the programmatic client

**Review by anayeaye — COMMENTED** [review]
Approve once @slesaad's suggested change to terraform.tfvars.tpl are accepted

File: .env.example
Comment (anayeaye) [inline .env.example:12]:
```suggestion
VEDA_WORKFLOWS_CLIENT_SECRET_ID=<Fill Me>
VEDA_PROGRAMMATIC_CLIENT_SECRET_ID=<Fill Me>
```

**Review by anayeaye — APPROVED** [review]

**Pull request merged**
